### PR TITLE
Alerting: Remove internal labels from prometheus compatible API responses

### LINKED ISSUE #45883: Unexpected labels for alerts in `/api/prometheus/grafana/api/v1/alerts`
**What happened**:

*This is for Prometheus alert-generator compliance, read more [here](https://github.com/prometheus/compliance/blob/main/alert_generator/specification.md). Using Grafana managed alert on a Prometheus datasource.*

A compliant behaviour would only expect the labels coming from the alerting rule query and labels from rule configuration attached to the alerts in `/api/prometheus/grafana/api/v1/alerts`.

Grafana attaches these additional labels which should not be present in that API:
* `__alert_rule_namespace_uid__`
* `__alert_rule_uid__`
* `datasource_uid`
* `ref_id`

**What you expected to happen**:

Those labels not present for the alerts in `/api/prometheus/grafana/api/v1/alerts`.

**Anything else we need to know?**:

This is mainly for how the API outputs the data. The internals of the alerting engine might not require any changes for this.

**Environment**:
- Grafana version: 8.4.2

### PR DESCRIPTION
**Pull request by gotjosh**
Fixes #45883

Grafana has two labels special labels used for internals `__alert_rule_uid__` and `__alert_rule_namespace_uid__`. These labels are only here for internal purposes and serve of no value to the consumers of the prometheus compatible API.

https://github.com/grafana/grafana/blob/e62e9904ee736fc04d661cd11735e221d16b1fc0/pkg/services/ngalert/state/cache.go#L86-L90

Furthermore, we even hide them on the front-end:

https://github.com/grafana/grafana/blob/e62e9904ee736fc04d661cd11735e221d16b1fc0/public/app/features/alerting/unified/components/AlertLabels.tsx#L9-L11

In this PR, I'm proposing we hide them by default and add an option for showing them if need be. I think the latter is useful for understanding and debugging what are the _actual_ factual labels that we currently hold.

### CODE REVIEW

File: pkg/services/ngalert/api/tooling/definitions/prom.go [hunk 1]
```diff
@@ -115,10 +115,10 @@ type Rule struct {
 	Name string `json:"name"`
 	// required: true
 	Query  string         `json:"query"`
-	Labels overrideLabels `json:"labels"`
```
Comment (gotjosh) [inline pkg/services/ngalert/api/tooling/definitions/prom.go:118]: drive-by fix due to https://github.com/prometheus/prometheus/blob/main/web/api/v1/api.go#L1183-L1185

File: pkg/services/ngalert/api/api_prometheus.go [hunk 2]
```diff
@@ -28,6 +28,14 @@ type PrometheusSrv struct {
 	store   store.RuleStore
 }
 
+const queryIncludeInternalLabels = "includeInternalLabels"
+
+// grafanaLabels are labels that grafana automatically include as part of the labelset.
+var grafanaLabels = map[string]struct{}{
+	ngmodels.RuleUIDLabel:      {},
+	ngmodels.NamespaceUIDLabel: {},
+}
```
Comment (yuri-tceretian) [inline pkg/services/ngalert/api/api_prometheus.go:37]: I think this list should be in the models and we should protect all APIs: remove from responses, fail requests that have those labels

**Review by yuri-tceretian — APPROVED** [review]

File: pkg/services/ngalert/api/api_prometheus.go (hunk 2 above)
Comment (gotjosh) [reply pkg/services/ngalert/api/api_prometheus.go]: Does https://github.com/grafana/grafana/pull/46548/commits/78d25fb783f4d50dcd0472b21ec08101b70721de address your main concern?

There's still the question of why is `Labels` public on both the `state` and `ngmodels` packages. I took a stab at refactoring it but it was certainly getting out of hand for this PR.

If the main concern about the boundary not being crossed on the API is addressed, I'm happy to follow up on the privatisation of `Labels` for both of these packages in the next PR.

Comment (gotjosh) [reply pkg/services/ngalert/api/api_prometheus.go]: Further discussed this with @yuri-tceretian offline -- I misunderstood his original comment but we both realised this is not an issue at this current state.

**Pull request merged**